Bug: udclient in Release

I believe we've found a bug on the udclient task when executing from a Release definition. 

Specifically, the check the udclient task makes on [line 39](https://github.com/IBM-UrbanCode/vsts-urbancode-deploy/blob/master/Extension/udClient/udClient.ts#L39) uses the variable `build.SourcesDirectory` as the comparison point to determine whether the udclient location was explicitly specified in the task. 

This works just fine from a Build Definition, but when the task is used in a Release definition the `build.SourcesDirectory` evaluates to `undefined`, resulting in that condition *always* being taken, even when the user did not explicitly specify a path to udclient.

I think instead that comparison could use the variable `System.DefaultWorkingDirectory` , as this will evaluate correctly on both build definitions (it has the same value as `Build.SourcesDirectory`) as well as in Release definitions. 

I've included some slightly scrubbed log output (with debug enabled) from the udclient task in one of our Release definitions for reference. I'd also be happen to open a PR with the change.

```sh
2018-07-24T00:47:28.6764353Z ##[debug]workingDirectory=***/a/_w/r2/a
2018-07-24T00:47:28.6776773Z ##[debug]udClientLocation=***/a/_w/r2/a
2018-07-24T00:47:28.6777639Z ##[debug]build.sourcesDirectory=undefined
2018-07-24T00:47:28.6778522Z ##[debug]udclient location specified explicitly by task: ***/a/_w/r2/a
2018-07-24T00:47:28.6783854Z ##[debug]resolving udclient.jar from udClientPath: ***/a/_w/r2/a
2018-07-24T00:47:28.6880636Z ##[error]resolved udclient.jar: ***/a/_w/r2/a/udclient.jar is not a file. Error: ENOENT:
```

--- a/Extension/udClient/task.json
+++ b/Extension/udClient/task.json
@@ -13,7 +13,7 @@
     "demands": [],
     "version": {
         "Major": 0,
-        "Minor": 121,
+        "Minor": 122,
         "Patch": 0
     },
     "groups": [

--- a/Extension/udClient/udClient.ts
+++ b/Extension/udClient/udClient.ts
@@ -36,7 +36,7 @@ var workingDir = tl.getInput('workingDirectory', true);
 
 var udClientPath; // loaded below
 var udClientLocation = tl.getInput('udClientLocation');
-if (udClientLocation != tl.getVariable('build.sourcesDirectory')) {
+if (udClientLocation != tl.getVariable('System.DefaultWorkingDirectory')) {
     //custom tool location for udclient was specified
     tl.debug('udclient location specified explicitly by task: ' + udClientLocation);
     udClientPath = udClientLocation;

--- a/package.json
+++ b/package.json
@@ -7,7 +7,8 @@
     "doc": "docs"
   },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "gulp": "gulp"
   },
   "repository": {
     "type": "git",

